tuna/miopen/miopen_lib.py

```diff
@@ -0,0 +1,368 @@
+                        required=False,
+                        help='Specify machine ids to use, comma separated')
+    parser.add_argument('-i',
+                        '--reset_interval',
+                        type=int,
+                        dest='reset_interval',
+                        required=False,
+                        help='Restart interval for job in hours.')
+
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument('--init_session',
+                       action='store_true',
+                       dest='init_session',
+                       help='Set up a new tuning session.')
+    group.add_argument(
+        '--fin_steps',
+        type=str,
+        dest='fin_steps',
+        help='Specify fin steps. Multiple steps should be comma separated.')
+    group.add_argument('--list_solvers',
+                       action='store_true',
+                       dest='list_solvers',
+                       help='List of solvers from the solver table')
+
+    # JD: implement the following two using fin_steps
+    group.add_argument('--update_solvers',
+                       dest='update_solvers',
+                       action='store_true',
+                       help='Update the list of solvers in the database')
+    group.add_argument('--update_applicability',
+                       dest='update_applicability',
+                       action='store_true',
+                       help='Update the applicability table in the database')
+    group.add_argument('-r',
+                       '--restart',
+                       dest='restart_machine',
+                       action='store_true',
+                       default=False,
+                       help='Restart machines')
+    group.add_argument('-s',
+                       '--status',
+                       dest='check_status',
+                       action='store_true',
+                       default=False,
+                       help='Check the status of machines')
+
+    group.add_argument('-d',
+                       '--docker_exec',
+                       dest='execute_docker_cmd',
+                       type=str,
+                       default=None,
+                       help='execute in a docker on each machine')
+    group.add_argument('-e',
+                       '--exec',
+                       dest='execute_cmd',
+                       type=str,
+                       default=None,
+                       help='execute on each machine')
+
+    parser.add_argument('-L',
+                        '--lib',
+                        dest='lib',
+                        help=argparse.SUPPRESS,
+                        default=Library.MIOPEN,
+                        choices=Library,
+                        type=Library)
+
+    args = parser.parse_args()
+    if len(sys.argv) == 3 and ('--lib' in sys.argv or
+                               '-L' in sys.argv) and 'miopen' in sys.argv:
+      parser.print_help()
+      sys.exit(-1)
```

Comment:
@cderb  @JehandadKhan  do you have a better idea on how to handle this? I basically want it to return the -H when user only puts in: "go_fish.py -L miopen" with no other args.